crates/ruff_python_parser/src/lexer.rs

```diff
@@ -51,6 +51,9 @@ use crate::{
 mod cursor;
 mod indentation;
 
+/// Maximum nesting level of f-string expressions.
+const MAX_FSTRING_EXPRESSION_NESTING: u32 = 3;
```

Comment:
Is there a technical reason why we want to limit the fstring nesting?

Reply:
No, PEP says that the lower bound is 2. While there's no upper bound, CPython implementation itself have an [upper bound of 3](https://github.com/python/cpython/blob/fd195092204aa7fc9f13c5c6d423bc723d0b3520/Parser/tokenizer.h#L43) and [here's](https://github.com/python/cpython/blob/fd195092204aa7fc9f13c5c6d423bc723d0b3520/Parser/tokenizer.c#L2650-L2652) where they throw an error if it goes beyond that. Currently, we have an upper bound of 2

https://github.com/astral-sh/ruff/blob/0cea4975fcfe09716c084c0a18d1b4c4cd9b8f05/crates/ruff_python_parser/src/string.rs#L420-L422

Reply:
I would remove the bound because different interpreter implementations may use different limits. However, it would be a good idea to have a lint rule that warns about too deeply nested FString depending on your target runtime.

Reply:
There's also the limitation of nested f-strings itself, so `f"foo {f"bar {f"baz"}"}"` which is also [limited to 150](https://github.com/python/cpython/blob/0cb0c238d520a8718e313b52cffc356a5a7561bf/Parser/tokenizer.h#L15). I'm avoiding that in our implementation for the same reason.